src/file-manager/fm-desktop-icon-view.c

```diff
@@ -708,11 +779,64 @@ action_change_background_callback (GtkAction *action,
 {
     g_assert (FM_DIRECTORY_VIEW (data));
 
-    caja_launch_application_from_command (gtk_widget_get_screen (GTK_WIDGET (data)),
-                                          _("Background"),
-                                          "mate-appearance-properties",
-                                          FALSE,
-                                          "--show-page=background", NULL);
+    if (GDK_IS_X11_DISPLAY (gdk_screen_get_display (gdk_screen_get_default())))
+        caja_launch_application_from_command (gtk_widget_get_screen (GTK_WIDGET (data)),
+                                              _("Background"),
+                                              "mate-appearance-properties",
+                                              FALSE,
+                                              "--show-page=background", NULL);
+    else
+    {
+        GtkWidget *dialog, *box;
+        gchar *filename;
+        GtkWidget *applybutton, *cancelbutton, *filebutton;
+        gint response;
+
+        filename = g_settings_get_string (background_settings, "picture-filename");
+
+        dialog = gtk_dialog_new ();
+        gtk_window_set_title (GTK_WINDOW (dialog), "Desktop Background Preferences");
```

Comment:
gtk_window_set_title (GTK_WINDOW (dialog), _("Desktop Background Preferences"));

Reply:
Included that change in latest force-push